packages/h5wasm/src/h5wasm-api.ts

```diff
@@ -75,12 +70,24 @@ export class H5WasmApi extends ProviderApi {
   }
 
   private async initFile(buffer: ArrayBuffer): Promise<H5WasmFile> {
-    await ready;
+    await h5wasmReady;
+
+    // Write file to Emscripten virtual file system
+    // https://emscripten.org/docs/api_reference/Filesystem-API.html#FS.writeFile
+    FS?.writeFile(this.filepath, new Uint8Array(buffer), { flags: 'w+' });
```

Comment:
@bmaranville any reason why `FS` can now be `null`?

Reply:
It is exported at the top level, and was previously initialized as null, but typed as Emscripten.FileSystemType.  This was caught by strict type checking and I updated the type.

This one is a little tricky, as the value of FS is replaced by the wasm Module.FS when the wasm is initialized, so if you await h5wasm.ready before using FS it will always be the right thing.  It's possible FS should be presented as a promise, instead.

Reply:
Right, I see. I'll replace `?.` with `!.` for now, then.